src/Countries/US/Washington/WashingtonWorkersCompensation/V20190101/WashingtonWorkersCompensation.php

```diff
@@ -3,28 +3,72 @@
+    public function __construct(Payroll $payroll)
     {
         parent::__construct($payroll);
-        $this->tax_information = $tax_information;
     }
 
     public function compute(Collection $tax_areas)
     {
-        $hourly_tax = $this->payroll->withholdTax($this->payroll->getShiftHoursWorked($tax_areas->first()->workGovernmentalUnitArea) * ($this->tax_information->employee_rate / 100));
+        $hourly = $this->payroll
+            ->getWages($tax_areas->first()->workGovernmentalUnitArea, WageType::SHIFT)
+            ->groupBy(function (Wage $wage) {
+                return $wage->getPosition();
+            })
+            ->map(function ($wages) {
+                $position = $wages->first()->getPosition();
+                $rate = $this->payroll->getWorkerCompRate('WA', $position);
+                $amount = $wages->sum(function (Wage $wage) {
+                            return $wage->getWorkTimeInMinutes();
+                }) / 60 * $rate->employee_rate / 100;
+                $earnings = $wages->sum(function (Wage $wage) {
+                        return $wage->getAmountInCents();
+                }) / 100 - $this->payroll->exempted_earnings;
 
-        if ($this->payroll->getStartDate()->weekOfMonth < 5 && $this->payroll->isSalariedWorker($tax_areas->first()->workGovernmentalUnitArea)) {
-            $salaried_tax = $this->payroll->withholdTax(40 * ($this->tax_information->employee_rate / 100));
+                return [
+                    'position' => $position,
+                    'amount' => $amount,
+                    'earnings' => $earnings,
+                ];
+            });
+
+
+        if ($this->payroll->getStartDate()->weekOfMonth < 5
+            && $this->payroll->isSalariedWorker($tax_areas->first()->workGovernmentalUnitArea)) {
+            $salary = $this->payroll
+                ->getWages($tax_areas->first()->workGovernmentalUnitArea, WageType::SALARY)
+                ->groupBy(function (Wage $wage) {
+                    return $wage->getPosition();
+                })
+                ->map(function ($wages) {
+                    $position = $wages->first()->getPosition();
+                    $rate = $this->payroll->getWorkerCompRate('WA', $position);
+                    return [
+                        'position' => $position,
+                        'amount' => ($wages->sum(function (Wage $wage) {
+                            return $wage->getWorkTimeInMinutes();
+                        }) / 60) * ($rate->employee_rate / 100) * 40,
+                        'earnings' => ($wages->sum(function (Wage $wage) {
+                            return $wage->getAmountInCents();
+                        })) / 100 - $this->payroll->exempted_earnings,
+                    ];
+                });
         } else {
-            $salaried_tax = 0;
+            $salary = collect([]);
+        }
+
+        // TODO: combine salary and hourly
+        if ($hourly->isNotEmpty()) {
```

Comment:
does this still need to be done? looks like if they have hourly and salary right now it will only return the hourly